src/config/common.rs

```diff
@@ -40,6 +40,34 @@ pub enum TransitionType {
     SlideEnd,
 }
 
+#[derive(Debug, Default, Deserialize, Clone, Copy)]
+#[serde(rename_all = "snake_case")]
+pub enum ModuleOrientation {
+    #[default]
+    #[serde(alias = "h")]
+    Horizontal,
+    #[serde(alias = "v")]
+    Vertical,
+}
+
+impl ModuleOrientation {
+    pub const fn to_angle(self) -> f64 {
+        match self {
+            Self::Horizontal => 0.0,
+            Self::Vertical => 90.0,
+        }
+    }
+}
+
+impl From<ModuleOrientation> for Orientation {
+    fn from(o: ModuleOrientation) -> Self {
+        match o {
+            ModuleOrientation::Horizontal => Orientation::Horizontal,
+            ModuleOrientation::Vertical => Orientation::Vertical,
```

Comment:
You can use the `Self` alias here:

```suggestion
            ModuleOrientation::Horizontal => Self::Horizontal,
            ModuleOrientation::Vertical => Self::Vertical,
```